roles/kubernetes-apps/ansible/tasks/main.yaml

```diff
@@ -5,7 +5,7 @@
     - {file: kubedns-rc.yml, type: rc}
     - {file: kubedns-svc.yml, type: svc}
   register: manifests
-  when: inventory_hostname == groups['kube-master'][0]
+  when: dns_mode != 'none' and inventory_hostname == groups['kube-master'][0]
```

Comment:
could we merge the `dns_mode='none'` with `dns_setup: false` some how ? this looks a bit confusing